taccsite_cms/settings.py

```diff
@@ -44,6 +44,7 @@ def gettext(s): return s
 
 # whether the session cookie should be secure (https:// only)
 SESSION_COOKIE_SECURE = True
+DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
```

Comment:
**Confirmation Request.**

1. This solves [warning about primary key since Django 3.2](https://dev.to/weplayinternet/upgrading-to-django-3-2-and-fixing-defaultautofield-warnings-518n)[^1], correct?
2. [This method may mean we need to run migrations…](https://dev.to/weplayinternet/upgrading-to-django-3-2-and-fixing-defaultautofield-warnings-518n#migrations) so do we need to worry about third-party plugin's own migration status?
3. Would we benefit from the ["No migrations please" route](https://dev.to/weplayinternet/upgrading-to-django-3-2-and-fixing-defaultautofield-warnings-518n#no-migrations-please)?

[^1]: `WARNINGS: blog.BlogPost: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.`

Reply:
I did a couple tests, even *without* the DEFAULT_AUTO_FIELD set there are still migrations that get generated when you run `makemigrations` in the main branch. I think this setting is just making something explicit that was set implicitly before.

Reply:
That makes sense. But my testing shows a difference in the migrations made — on Core-CMS (and tup-ui) — using `main` versus this branch (and image from it).

I will try `DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'` to see if it reduces (hopefully eliminates) the diff in migrations.

<sup>If the warnings for `djangocms_blog` remain, I do not worry, because they will be removed when I update `djangocms_blog` because https://github.com/nephila/djangocms-blog/pull/741.</sup>

<sup>I fear making migrations in our app for 3rd-party apps, because I do not understand migrations well enough to administer what I see as "custom migrations".</sup>

<details><summary>Differences</summary>

1. New branch migrations removes all warnings.*
2. New branch migrations show new `Alter field id on …` migrations which alters field to be `…BigAutoField`.*

<sup>* The `id` fields on `djangocms_blog` would be altered to `…BigAutoField`, hence the warnings being gone.</sup>

</details>

<details><summary>Testing</summary>

1. check out `main` branch
2. `build` new image
4. `start` new image
5. in Docker, `rm` all the migrations files committed by any past `makemigrations` testing <sub>(this step is admittedly difficult to verify)</sub>
8. in Docker, `rm` all the migrations files committed by any past `makemigrations` testing
6. in Docker, `… makemigrations`
7. save output
8. check out **new** branch
    - Core-CMS `deps/python-3.11`
    - TUP-UI:`chore/collectstatic-on-build`
9. in Docker, `… makemigrations`
10. save output
11. compare output

</details>

<details><summary>Resources</summary>

- [Core main - makemigrations.log](https://github.com/TACC/Core-CMS/files/13955991/Core.main.-.makemigrations.log)
- [Core python-3.11 - makemigrations.log](https://github.com/TACC/Core-CMS/files/13955989/Core.python-3.11.-.makemigrations.log)
- [TUP main - makemigrations.log](https://github.com/TACC/Core-CMS/files/13955988/TUP.main.-.makemigrations.log)
- [TUP collectstatic-on-build - makemigrations.log](https://github.com/TACC/Core-CMS/files/13955990/TUP.collectstatic-on-build.-.makemigrations.log)

</details>

Reply:
Using `AutoField` instead does **not** add migrations and **still** removes the warnings. I think this is preferable.

```suggestion
DEFAULT_AUTO_FIELD = 'django.db.models.AutoField'
```

<details><summary>Resources</summary>

Compare resources 2 and 3. Notice that there are **no** changes to`id` fields. (Changes to `id` fields in 2 are altering field to be `BigAutoField`.)

1. [Core.main.-.makemigrations.log](https://github.com/TACC/Core-CMS/files/13980392/Core.main.-.makemigrations.log)
2. [Core.python-3.11.-BigAutoField-.makemigrations.log](https://github.com/TACC/Core-CMS/files/13980393/Core.python-3.11.-.makemigrations.log)
3. [Core.python-3.11.-AutoField-.makemigrations.log](https://github.com/TACC/Core-CMS/files/13980394/Core.python-3.11.-EDIT-.makemigrations.log)

</details>

Reply:
I retract my request and bury my concern.

@wesleyboar, this is why you can let this go:

- The migrations for third-party apps are local/transient. They won't be run again.
- The end result is still `BigAutoField`. Trust that Django is smart enough to see that.
- If conflicts occur, there is a `… makemigrations --merge <app>` that might help.